x/authority/keeper/keeper.go

```diff
@@ -140,3 +188,22 @@ func (k Keeper) initGasPrices(ctx sdk.Context) {
 		k.gpk.SetMinimumGasPrices(gps.String())
 	})
 }
+
+func (k Keeper) replaceAuthority(ctx sdk.Context, authority, newAuthority sdk.AccAddress) (*sdk.Result, error) {
+	if err := k.ValidateAuthority(ctx, authority); err != nil {
+		return nil, err
+	}
+
+	authoritySet := k.GetAuthoritySet(ctx)
+
+	formerAuthorityAddress := authoritySet.FormerAddress
+	pastTransitionPeriod := authoritySet.LastModified.Add(types.AuthorityTransitionDuration).Before(ctx.BlockTime())
+	if pastTransitionPeriod || formerAuthorityAddress == "" {
+		// Change the fallback address past the transition period
+		formerAuthorityAddress = authoritySet.Address
+	}
```

Comment:
This would IMO be more readable if it just called `getAuthority` and then did the "keep former authority if it exists"-logic based on the results from that function. Would also serve to keep the timestamp check in a single place within the keeper.

Reply:
Brilliance touch! Thx a lot!